api/app/controllers/api/v1/applications_controller.rb

```diff
@@ -47,6 +54,22 @@ def update
             end
         end
 
+        # /applications/delete or /applications/:id/delete
+        def delete
+            params.require(:id)
+            application = Application.find(params[:id])
+            if application
+                applicant_data_for_matching = ApplicantDataForMatching.find_by(
+                    application_id: params[:id])
+                applicant_data_for_matching.destroy!
```

Comment:
Also are we guaranteed applicant_data_for_matching? If there isn't one, won't `destroy` fail and muck up everything down stream? Should we gate this with an `applicant_data_for_matching.destroy! if applicant_data_for_matching`?

Reply:
It should be guaranteed because the controller is assuming that our tables are always created via the API routes. If there was an error in the `applicant_data_for_matching` creation, it should have shown in the error message and the corresponding `application` should have been deleted.

Reply:
I meant it in the sense that couldn't we have an application without an applicant data for matching?

Reply:
Not really. @siefkenj's explanation for this was that the `application` and `applicant_data_for_matching` are really meant to be the same table and that they are only separated for semantics.